src/metatracer.rs

```diff
@@ -54,68 +94,131 @@ const PHASE_COUNTING: u32 = 0b10 << 30; // The value specifies the current hot c
 /// `Location`. For interpreters that can't (or don't want) to be as selective, a simple (if
 /// moderately wasteful) mechanism is for every bytecode or AST node to have its own `Location`
 /// (even for bytecodes or nodes that can't be control points).
+///
+/// Location is shareable amongst threads.
 pub struct Location {
-    pack: AtomicU32
+    pack: Arc<AtomicU32>,
 }
 
 impl Location {
     /// Create a fresh Location suitable for passing to `MetaTracer::control_point`.
     pub fn new() -> Self {
-        Self { pack: AtomicU32::new(PHASE_COUNTING) }
+        Self { pack: Arc::new(AtomicU32::new(PHASE_COUNTING)) }
     }
 }
 
 /// A meta-tracer.
 pub struct MetaTracer {
-    hot_threshold: HotThreshold
+    tracer: Box<Tracer>,                // A `Tracer` from which to make `ThreadTracers`.
+    hot_threshold: HotThreshold,
+    code_cache: Arc<RwLock<Vec<Code>>>, // Compiled traces. Indices serve as unique identifiers.
```

Comment:
We really want a lockless structure here I suspect: `Arc<RwLoc>>` is not going to be very fast.

Reply:
Eventually yeah.